src/main/app/src/app/admin/smart-documents.service.ts

```diff
@@ -192,28 +128,24 @@ export class SmartDocumentsService {
       .filter(Boolean);
   };
 
-  flattenNestedGroupsToRootGroups = (
-    data: MappedSmartDocumentsTemplateGroupWithParentId[],
-  ): MappedSmartDocumentsTemplateFlattenedGroupWithParentId[] => {
-    return data.flatMap((item) => {
-      const rootItemWithoutGroups = {
-        id: item.id,
-        name: item.name,
-        templates: item.templates,
-      };
+  flattenNestedGroups(
+    groups: MappedSmartDocumentsTemplateGroupWithParentId[],
+  ): MappedSmartDocumentsTemplateFlattenedGroupWithParentId[] {
+    const result = [];
+
+    function traverse(group) {
+      const { id, name, templates = [] } = group;
```

Comment:
I don't get why we only grab these three properties and not the `groups`. That is used a bit further down directly from the `group`...

Reply:
so the reason is that these 3 fields are used separately from the `groups` field, and this way that is more obvious? I don't agree with that observation. If we use all these fields, let's get them consistently from the `group` that's passed in, maybe destruct it straight in the function parameter definition? Then we don't even need `group` any more.

Reply:
Just trying this out locally and I also see there is no type-safety on `result` nor on the `traverse` method.... is that normal? And why?

Reply:
I would go for this `traverse` method definition:
```
function traverse({ id, name, templates = [], groups }: MappedSmartDocumentsTemplateGroupWithParentId) {
```

Reply:
I suggest the following:
```suggestion
function traverse({ id, name, templates = [], groups }: MappedSmartDocumentsTemplateGroupWithParentId) {
```